ranking/src/ts_system/normal.rs

```diff
@@ -0,0 +1,191 @@
+extern crate overload;
+
+use overload::overload;
+use std::ops;
+
+use statrs::function::erf::erfc;
+use std::f64::consts::PI;
+use std::f64::INFINITY;
+
+const PREC: f64 = 1e-4;
+
+#[derive(Clone, Debug)]
+pub struct Gaussian {
+    pub mu: f64,
+    pub sigma: f64,
+}
+
+#[allow(unused)]
+pub const ZERO: Gaussian = Gaussian { mu: 0., sigma: 0. };
+#[allow(unused)]
+pub const ONE: Gaussian = Gaussian {
+    mu: 0.,
+    sigma: INFINITY,
+};
+
+overload!((a: ?Gaussian) + (b: ?Gaussian) -> Gaussian {
+    Gaussian {
+        mu: a.mu + b.mu,
+        sigma: (a.sigma.powi(2) + b.sigma.powi(2)).sqrt(),
+    }
+});
+
+overload!((a: &mut Gaussian) += (b: ?Gaussian) {
+    a.mu += b.mu;
+    a.sigma = (a.sigma.powi(2) + b.sigma.powi(2)).sqrt();
+});
+
+overload!((a: ?Gaussian) - (b: ?Gaussian) -> Gaussian {
+    Gaussian {
+        mu: a.mu - b.mu,
+        sigma: (a.sigma.powi(2) + b.sigma.powi(2)).sqrt(),
+    }
+});
+
+overload!((a: &mut Gaussian) -= (b: ?Gaussian) {
+    a.mu -= b.mu;
+    a.sigma = (a.sigma.powi(2) + b.sigma.powi(2)).sqrt();
+});
+
+overload!(-(a: &mut Gaussian) -> Gaussian {
+    Gaussian {
+        mu: -a.mu,
+        sigma: a.sigma,
+    }
+});
+
+overload!((a: ?Gaussian) * (b: ?f64) -> Gaussian {
+    Gaussian {
+        mu: a.mu * b,
+        sigma: a.sigma * b.abs(),
+    }
+});
+
+overload!((a: &mut Gaussian) *= (b: ?f64) {
+    a.mu *= b;
+    a.sigma *= b.abs();
+});
+
+overload!((a: ?Gaussian) / (b: ?f64) -> Gaussian {
+    Gaussian {
+        mu: a.mu / b,
+        sigma: a.sigma / b.abs(),
+    }
+});
+
+overload!((a: &mut Gaussian) /= (b: ?f64) {
+    a.mu /= b;
+    a.sigma /= b.abs();
+});
+
+overload!((a: ?Gaussian) * (b: ?Gaussian) -> Gaussian {
+    if a.sigma.is_infinite() {
+        return b.clone();
```

Comment:
`clippy` warning: it looks like `a` and `b` don't need to be `clone()`d here, since the "originals" are dropped after this point.